// eventregistry/dist/returnInfo.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var _ = require("lodash");
var ReturnInfoFlagsBase = /** @class */ (function () {
    function ReturnInfoFlagsBase() {
        this.data = {};
    }
    ReturnInfoFlagsBase.prototype.setFlag = function (key, obj, defaultValue) {
        if (_.has(obj, key)) {
            this.setProperty("Include" + this.type + _.upperFirst(key), obj[key], defaultValue);
        }
    };
    ReturnInfoFlagsBase.prototype.setValue = function (key, obj, defaultValue, skipKeyMod) {
        if (skipKeyMod === void 0) { skipKeyMod = false; }
        if (_.has(obj, key)) {
            var constructedKey = skipKeyMod ? _.upperFirst(key) : this.type + _.upperFirst(key);
            this.setProperty(constructedKey, obj[key], defaultValue);
        }
    };
    ReturnInfoFlagsBase.prototype.getProperties = function (prefix) {
        if (prefix === void 0) { prefix = ""; }
        return _.mapKeys(this.data, function (value, key) {
            if (_.startsWith(_.toLower(key), _.toLower(prefix))) {
                return _.camelCase(key);
            }
            else {
                return _.camelCase(prefix + key);
            }
        });
    };
    ReturnInfoFlagsBase.prototype.setProperty = function (key, value, defaultValue) {
        if (value !== defaultValue) {
            _.set(this.data, key, value);
        }
    };
    return ReturnInfoFlagsBase;
}());
exports.ReturnInfoFlagsBase = ReturnInfoFlagsBase;
var ReturnInfo = /** @class */ (function () {
    // Accepts all InfoFlags
    function ReturnInfo(_a) {
        var _b = _a === void 0 ? {} : _a, _c = _b.articleInfo, articleInfo = _c === void 0 ? new ArticleInfoFlags() : _c, _d = _b.eventInfo, eventInfo = _d === void 0 ? new EventInfoFlags() : _d, _e = _b.sourceInfo, sourceInfo = _e === void 0 ? new SourceInfoFlags() : _e, _f = _b.categoryInfo, categoryInfo = _f === void 0 ? new CategoryInfoFlags() : _f, _g = _b.conceptInfo, conceptInfo = _g === void 0 ? new ConceptInfoFlags() : _g, _h = _b.locationInfo, locationInfo = _h === void 0 ? new LocationInfoFlags() : _h, _j = _b.storyInfo, storyInfo = _j === void 0 ? new StoryInfoFlags() : _j, _k = _b.conceptClassInfo, conceptClassInfo = _k === void 0 ? new ConceptClassInfoFlags() : _k, _l = _b.conceptFolderInfo, conceptFolderInfo = _l === void 0 ? new ConceptFolderInfoFlags() : _l;
        this.articleInfo = articleInfo;
        this.eventInfo = eventInfo;
        this.sourceInfo = sourceInfo;
        this.categoryInfo = categoryInfo;
        this.conceptInfo = conceptInfo;
        this.locationInfo = locationInfo;
        this.storyInfo = storyInfo;
        this.conceptClassInfo = conceptClassInfo;
        this.conceptFolderInfo = conceptFolderInfo;
    }
    ReturnInfo.prototype.getParams = function (prefix) {
        return _.extend({}, this.articleInfo.getProperties(prefix), this.eventInfo.getProperties(prefix), this.sourceInfo.getProperties(prefix), this.categoryInfo.getProperties(prefix), this.conceptInfo.getProperties(prefix), this.locationInfo.getProperties(prefix), this.storyInfo.getProperties(prefix), this.conceptClassInfo.getProperties(prefix), this.conceptFolderInfo.getProperties(prefix));
    };
    return ReturnInfo;
}());
exports.ReturnInfo = ReturnInfo;
var ArticleInfoFlags = /** @class */ (function (_super) {
    __extends(ArticleInfoFlags, _super);
    function ArticleInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Article";
        _this.setValue("bodyLen", params, 300);
        _this.setFlag("basicInfo", params, true);
        _this.setFlag("title", params, true);
        _this.setFlag("body", params, true);
        _this.setFlag("url", params, true);
        _this.setFlag("eventUri", params, true);
        _this.setFlag("concepts", params, false);
        _this.setFlag("categories", params, false);
        _this.setFlag("links", params, false);
        _this.setFlag("videos", params, false);
        _this.setFlag("image", params, false);
        _this.setFlag("socialScore", params, false);
        _this.setFlag("sentiment", params, false);
        _this.setFlag("location", params, false);
        _this.setFlag("dates", params, false);
        _this.setFlag("extractedDates", params, false);
        _this.setFlag("duplicateList", params, false);
        _this.setFlag("originalArticle", params, false);
        _this.setFlag("storyUri", params, false);
        return _this;
    }
    return ArticleInfoFlags;
}(ReturnInfoFlagsBase));
exports.ArticleInfoFlags = ArticleInfoFlags;
var StoryInfoFlags = /** @class */ (function (_super) {
    __extends(StoryInfoFlags, _super);
    function StoryInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Story";
        _this.setFlag("basicStats", params, true);
        _this.setFlag("location", params, true);
        _this.setFlag("date", params, false);
        _this.setFlag("title", params, false);
        _this.setFlag("summary", params, false);
        _this.setFlag("concepts", params, false);
        _this.setFlag("categories", params, false);
        _this.setFlag("medoidArticle", params, false);
        _this.setFlag("infoArticle", params, false);
        _this.setFlag("commonDates", params, false);
        _this.setFlag("socialScore", params, false);
        _this.setValue("imageCount", params, 0);
        return _this;
    }
    return StoryInfoFlags;
}(ReturnInfoFlagsBase));
exports.StoryInfoFlags = StoryInfoFlags;
var EventInfoFlags = /** @class */ (function (_super) {
    __extends(EventInfoFlags, _super);
    function EventInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Event";
        _this.setFlag("title", params, true);
        _this.setFlag("summary", params, true);
        _this.setFlag("articleCounts", params, true);
        _this.setFlag("concepts", params, true);
        _this.setFlag("categories", params, true);
        _this.setFlag("location", params, true);
        _this.setFlag("date", params, true);
        _this.setFlag("commonDates", params, false);
        _this.setFlag("infoArticle", params, false);
        _this.setFlag("stories", params, false);
        _this.setFlag("socialScore", params, false);
        _this.setValue("imageCount", params, 0);
        return _this;
    }
    return EventInfoFlags;
}(ReturnInfoFlagsBase));
exports.EventInfoFlags = EventInfoFlags;
var SourceInfoFlags = /** @class */ (function (_super) {
    __extends(SourceInfoFlags, _super);
    function SourceInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Source";
        _this.setFlag("title", params, true);
        _this.setFlag("description", params, false);
        _this.setFlag("location", params, false);
        _this.setFlag("ranking", params, false);
        _this.setFlag("image", params, false);
        _this.setFlag("articleCount", params, false);
        _this.setFlag("sourceMedia", params, false);
        _this.setFlag("sourceGroups", params, false);
        return _this;
    }
    return SourceInfoFlags;
}(ReturnInfoFlagsBase));
exports.SourceInfoFlags = SourceInfoFlags;
var CategoryInfoFlags = /** @class */ (function (_super) {
    __extends(CategoryInfoFlags, _super);
    function CategoryInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Category";
        _this.setFlag("parentUri", params, false);
        _this.setFlag("childrenUris", params, false);
        _this.setFlag("trendingScore", params, false);
        _this.setFlag("trendingHistory", params, false);
        _this.setValue("trendingSource", params, "news");
        return _this;
    }
    return CategoryInfoFlags;
}(ReturnInfoFlagsBase));
exports.CategoryInfoFlags = CategoryInfoFlags;
var ConceptInfoFlags = /** @class */ (function (_super) {
    __extends(ConceptInfoFlags, _super);
    function ConceptInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Concept";
        _this.setValue("type", params, "concepts");
        _this.setValue("lang", params, "eng");
        _this.setFlag("label", params, true);
        _this.setFlag("synonyms", params, false);
        _this.setFlag("image", params, false);
        _this.setFlag("description", params, false);
        _this.setFlag("conceptClassMembership", params, false);
        _this.setFlag("conceptClassMembershipFull", params, false);
        _this.setFlag("trendingScore", params, false);
        _this.setFlag("trendingHistory", params, false);
        _this.setFlag("totalCount", params, false);
        _this.setValue("trendingSource", params, "news");
        _this.setValue("maxConceptsPerType", params, 20, true);
        return _this;
    }
    return ConceptInfoFlags;
}(ReturnInfoFlagsBase));
exports.ConceptInfoFlags = ConceptInfoFlags;
var LocationInfoFlags = /** @class */ (function (_super) {
    __extends(LocationInfoFlags, _super);
    function LocationInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "Location";
        _this.setFlag("label", params, true);
        _this.setFlag("wikiUri", params, false);
        _this.setFlag("geoNamesId", params, false);
        _this.setFlag("population", params, false);
        _this.setFlag("geoLocation", params, false);
        _this.setFlag("countryArea", params, false);
        _this.setFlag("countryDetails", params, false);
        _this.setFlag("countryContinent", params, false);
        _this.setFlag("placeFeatureCode", params, false);
        _this.setFlag("placeCountry", params, true);
        return _this;
    }
    return LocationInfoFlags;
}(ReturnInfoFlagsBase));
exports.LocationInfoFlags = LocationInfoFlags;
var ConceptClassInfoFlags = /** @class */ (function (_super) {
    __extends(ConceptClassInfoFlags, _super);
    function ConceptClassInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "ConceptClass";
        _this.setFlag("parentLabels", params, true);
        _this.setFlag("concepts", params, false);
        return _this;
    }
    return ConceptClassInfoFlags;
}(ReturnInfoFlagsBase));
exports.ConceptClassInfoFlags = ConceptClassInfoFlags;
var ConceptFolderInfoFlags = /** @class */ (function (_super) {
    __extends(ConceptFolderInfoFlags, _super);
    function ConceptFolderInfoFlags(params) {
        if (params === void 0) { params = {}; }
        var _this = _super.call(this) || this;
        _this.type = "ConceptFolder";
        _this.setFlag("definition", params, true);
        _this.setFlag("owner", params, false);
        return _this;
    }
    return ConceptFolderInfoFlags;
}(ReturnInfoFlagsBase));
exports.ConceptFolderInfoFlags = ConceptFolderInfoFlags;
